File: ios/LegalTalkIndia/Data/Model/Case.swift
import Foundation

struct CaseDocument: Codable, Identifiable {
    let id: String
    let name: String
    let url: String
    let uploadedBy: String // "client", "lawyer"
    let uploadedAt: String
}

struct Case: Codable, Identifiable {
    let id: String
    let clientId: String
    let clientName: String
    let lawyerId: String?
    let lawyerName: String?
    let title: String
    let description: String
    let category: String
    let status: String // "searching", "ongoing", "closed"
    let documents: [CaseDocument]
    let createdAt: String
}

File: android/app/src/main/java/com/legaltalk/india/data/model/Case.kt
package com.legaltalk.india.data.model

data class CaseDocument(
    val id: String,
    val name: String,
    val url: String,
    val uploadedBy: String, // "client", "lawyer"
    val uploadedAt: String
)

data class Case(
    val id: String,
    val clientId: String,
    val clientName: String,
    val lawyerId: String? = null,
    val lawyerName: String? = null,
    val title: String,
    val description: String,
    val category: String,
    val status: String, // "searching", "ongoing", "closed"
    val documents: List<CaseDocument> = emptyList(),
    val createdAt: String
)
